lib/datadog/appsec/monitor/gateway/watcher.rb

```diff
@@ -44,10 +44,10 @@ def watch_user_id(gateway = Instrumentation.gateway)
                     end
                   end
 
-                  _result, block = Monitor::Reactive::SetUser.publish(op, user)
+                  block = Monitor::Reactive::SetUser.publish(op, user)
                 end
 
-                throw(Datadog::AppSec::Ext::INTERRUPT, [nil, [:block, event]]) if block
+                throw(Datadog::AppSec::Ext::INTERRUPT, [nil, [[:block, event]]]) if block
```

Comment:
Curious about the double array here: was it missing before or is that an intended API change?

Reply:
In was missing 😨

Reply:
While missing the functionality still worked because we did not use the values 

```ruby
[:block, {}].any? {|action, _event| action == :block }
# => true
```

We now care about the returned value with the code changes on the request middleware. We use `find` rather than `any?`

```ruby
blocked_event = request_response.find { |action, _options| action == :block }
```


Old code
```ruby
blocked_event = [:block, {}].find {|action, _event| action == :block }
#=> :block
```

New Code
```ruby
blocked_event = [[:block, {}]].find {|action, _event| action == :block }
# => [:block, {}]
```